src/workerd/api/web-socket.c++

```diff
@@ -531,7 +531,10 @@ void WebSocket::send(jsg::Lock& js, kj::OneOf<kj::Array<byte>, kj::String> messa
 
     KJ_UNREACHABLE;
   }();
-  outgoingMessages->insert(GatedMessage{kj::mv(maybeOutputLock), kj::mv(msg)});
+
+  auto n = autoResponseStatus.pendingAutoResponseQueue.size() - autoResponseStatus.queuedAutoResponses;
```

Comment:
Can you explain the difference between `GatedMessage::pendingAutoResponses`, and `AutoResponse::queuedAutoResponses`?

Reply:
`GatedMessage::pendingAutoResponses` stores the number of `AutoResponse::queuedAutoResponses` that will be pumped before sending the current `GatedMessage`, guaranteeing order. `AutoResponse::queuedAutoResponses` stores the queue of raw `autoResponse` messages waiting to be pumped. At the end of the `pump` cycle, any pending messages stored in `AutoResponse::queuedAutoResponses` will be pumped.